extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/spi/MutinyWrapperHandlerService.java

```diff
@@ -0,0 +1,114 @@
+package io.quarkus.smallrye.graphql.runtime.spi;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+
+import org.dataloader.BatchLoaderEnvironment;
+import org.eclipse.microprofile.graphql.GraphQLException;
+
+import graphql.execution.DataFetcherResult;
+import graphql.schema.DataFetchingEnvironment;
+import io.smallrye.graphql.SmallRyeGraphQLServerMessages;
+import io.smallrye.graphql.schema.model.Field;
+import io.smallrye.graphql.schema.model.Wrapper;
+import io.smallrye.graphql.spi.WrapperHandlerService;
+import io.smallrye.graphql.spi.datafetcher.AbstractWrapperHandlerService;
+import io.smallrye.graphql.transformation.AbstractDataFetcherException;
+import io.smallrye.mutiny.Uni;
+
+/**
+ * Handle Async calls with Uni from Mutiny
+ */
+public class MutinyWrapperHandlerService extends AbstractWrapperHandlerService {
+
+    private static final List<String> FOR_CLASSES = Arrays.asList(new String[] {
+            Uni.class.getName()
+    });
+
+    @Override
+    public String getName() {
+        return "Mutiny";
+    }
+
+    @Override
+    public List<String> forClasses() {
+        return FOR_CLASSES;
+    }
+
+    @Override
+    public WrapperHandlerService newInstance() {
+        return new MutinyWrapperHandlerService();
+    }
+
+    @Override
+    protected <T> T invokeAndTransform(DataFetchingEnvironment dfe, DataFetcherResult.Builder<Object> resultBuilder,
+            Object[] transformedArguments) throws AbstractDataFetcherException, Exception {
+        Uni<?> uni = reflectionHelper.invoke(transformedArguments);
+
+        return (T) uni
+                .onItemOrFailure().transform((result, throwable) -> {
+
+                    if (throwable != null) {
+                        if (throwable instanceof CompletionException || shouldUnwrap(throwable)) {
```

Comment:
Uni won't throw a CompletionException, it's not a CS.

Reply:
I'll remove